src/components/CatalogLayersLegendItems.vue

```diff
@@ -83,14 +83,74 @@ export default {
       legendurl.loading = false;
     },
 
+    /**
+     * @FIXME add parameter types
+     * @FIXME add return type
+     * @FIXME missing return value when `catalogLayer` is not defined
+     * 
+     * @returns {string}
+     */
```

Comment:
Check these above comments before merge.

As for the rest it seems to me to be fine 👌